### ISSUE TITLE
Need help in marketing, content hunting for Awesome J2ME

### ISSUE THREAD

**Issue by hstsethi** [hstsethi/awesome-j2me#3]
## About Awesome J2ME 

Introducing [Awesome
J2ME](https://github.com/hstsethi/awesome-j2me), an Awesome list that I have been working on for more than a year. The goal of this project is to create a comprehensive resource that will be helpful to the whole community. Not only developers, video gamers, hobbyists but anyone who is interested in J2ME.

I managed to collect a diverse set of resources, and submit a [pull request](https://github.com/sindresorhus/awesome/pull/3115) that passed all the checks in first try.  But sadly, due to small community size, most communities being hosted on proprietary platforms,  lack of resources, exams and abandonment of J2ME by Oracle, I was not able to completely fulfill my goal.

So I would like to request you to help me in achieving this goal. Pull requests are always welcome. If you can't think of any good resource, simply sharing and giving it a star on Github is more than enough. 

Please forward this message to other communities about J2ME. Since I don't have Discord, Reddit, I can't reach them.

This was originally published on my [blog](https://hstsethi.vercel.app/posts/programming/awesome-j2me)

Thank you.

**Comment by 3dcinetv** [comment]
I want to contribute. I have a 30K follower channel.
I am also about to release a modern exporter for .m3g (3D) format for J2ME. The 2005 blender 2.49 was never ported to modern systems. 
Let me know if we can make this collab happen.

**Comment by hstsethi** [comment]
> I want to contribute. I have a 30K follower channel. I am also about to release a modern exporter for .m3g (3D) format for J2ME. The 2005 blender 2.49 was never ported to modern systems. Let me know if we can make this collab happen.

Yes, that would be awesome! Can we continue in private? Perhaps email or something else if you prefer.

BTW, the tutorials on your YT channel are really good.

**Comment by 3dcinetv** [comment]
awesome! 
you can go into my youtube channel>"more"> profile> Click on (show) email. 
Once you contact me, I'll share the progress. Maybe you know more about .m3g than I do.

**Comment by hstsethi** [comment]
> awesome! you can go into my youtube channel>"more"> profile> Click on (show) email. Once you contact me, I'll share the progress. Maybe you know more about .m3g than I do.

Youtube did not allow me to see your email without signing in. But I do not have a Google account, so I sent an email to the one listed on your Github. Hope you got that message.  

Also I have checked your repo on this. So do you want us to work on that Python plugin? What specific should we do?

**Comment by BaalNetbek** [comment]
Many basic examples: https://web.archive.org/web/20121102161050/http://www.java-tips.org/java-me-tips/midp/page3.html


J2ME performance tips: https://www.javaperformancetuning.com/tips/j2me.shtml#REF18 (https://web.archive.org/web/20050223040231/https://www.javaperformancetuning.com/tips/j2me.shtml#REF25)
General old java optimization tips: https://web.archive.org/web/20050215222448/http://www-2.cs.cmu.edu/~jch/java/
Basic java preformance good practices: https://web.archive.org/web/19970607121931/http://www.javaworld.com/javaworld/jw-04-1997/jw-04-optimize.html 
The JavaTM Virtual Machine Specification Second Edition: https://web.archive.org/web/19990508124421/http://java.sun.com/docs/books/vmspec/2nd-edition/html/VMSpecTOC.doc.html

**Comment by BaalNetbek** [comment]
Mostly development related tips and tricks from Sony Ericsson: https://web.archive.org/web/20061006072043/http://developer.sonyericsson.com/site/global/techsupport/tipstrickscode/java/p_java.jsp

**Comment by hstsethi** [comment]
> Mostly development related tips and tricks from Sony Ericsson: https://web.archive.org/web/20061006072043/http://developer.sonyericsson.com/site/global/techsupport/tipstrickscode/java/p_java.jsp

Thank you for these recommendations. I am noting and eyeing them down, and will do a commit for them shortly.

**Comment by hstsethi** [comment]
@BaalNetbek Thank you once again. I have incorporated your suggestions. I have added your name in Co-Author field but it won't be recognized until the email is given.  If you want to get credit, you can share, even the Github generated "no reply" one will work. 

But if you not comfortable, that is okay too.

**Comment by hourianto** [comment]
Hi, I have a Vineflower fork specifically focused at decompiling J2ME software, with a focus on making sure that the code is 1) semantically equivalent to the original bytecode and 2) can be recompiled back directly without any modifications. It obviously doesn't achieve that for all the jars out there, but together with LLM agents (mainly GPT 5.3/5.4/5.5) I've made a lot of progress and tens of games that either have semantic bugs or don't compile from the decompiled source with Vineflower, do compile with Sporeflower. https://github.com/hourianto/sporeflower


I know that the README and the page is quite empty, I should probably add some examples/comparisons, but honestly I don't know if it's possible to do it in a copyright-respecting way to the original .jars. Although I guess I could take some open-source J2ME software to try.

You can see that I filed 10+ valid bugs to Vineflower, all of which are fixed in my fork and were found by decompiling real J2ME projects: https://github.com/Vineflower/vineflower/issues?q=is%3Aissue%20state%3Aopen%20author%3Ahourianto. My fork has fixes to far more issues than just these, since a lot of them can't be represented with pure Java, but instead only J2ME-era weird bytecode.


As full disclosure - it's exclusively developed by LLM agents, but I don't let them blindly commit anything, I do re-reviews and sometimes spot problems or just bad places in their code. This message isn't written by an LLM though.

If you want, I could show some more work in private, I currently have a tool (I should release it publicly later) that allows easy remapping of J2ME jars with stripped names (most of the ones in the wild are like that) by writing Java-style definitions which are then bytecode-verified and converted into Tiny v2 mappings (common machine-readable remapping format, which my fork has support for)

**Comment by hstsethi** [comment]
> Hi, I have a Vineflower fork specifically focused at decompiling J2ME software, with a focus on making sure that the code is 1) semantically equivalent to the original bytecode and 2) can be recompiled back directly without any modifications. It obviously doesn't achieve that for all the jars out there, but together with LLM agents (mainly GPT 5.3/5.4/5.5) I've made a lot of progress and tens of games that either have semantic bugs or don't compile from the decompiled source with Vineflower, do compile with Sporeflower. https://github.com/hourianto/sporeflower
> 
> I know that the README and the page is quite empty, I should probably add some examples/comparisons, but honestly I don't know if it's possible to do it in a copyright-respecting way to the original .jars. Although I guess I could take some open-source J2ME software to try.
> 
> You can see that I filed 10+ valid bugs to Vineflower, all of which are fixed in my fork and were found by decompiling real J2ME projects: https://github.com/Vineflower/vineflower/issues?q=is%3Aissue%20state%3Aopen%20author%3Ahourianto. My fork has fixes to far more issues than just these, since a lot of them can't be represented with pure Java, but instead only J2ME-era weird bytecode.
> 
> As full disclosure - it's exclusively developed by LLM agents, but I don't let them blindly commit anything, I do re-reviews and sometimes spot problems or just bad places in their code. This message isn't written by an LLM though.
> 
> If you want, I could show some more work in private, I currently have a tool (I should release it publicly later) that allows easy remapping of J2ME jars with stripped names (most of the ones in the wild are like that) by writing Java-style definitions which are then bytecode-verified and converted into Tiny v2 mappings (common machine-readable remapping format, which my fork has support for)

Thank you for your work on Sporeflower! I have added it to the list. 

And yes, comparison using open source software or documentation examples might be the best approach. There are many in the list.

Yes, I would love to see that tool if you are open to.
